server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationStatsIT.java

```diff
@@ -47,7 +47,12 @@ public void testSegmentReplicationStatsResponse() throws Exception {
             .actionGet();
         // Verify API Response
         assertThat(response.shardSegmentReplicationStates().size(), equalTo(SHARD_COUNT));
-        assertThat(response.shardSegmentReplicationStates().get(INDEX_NAME).get(0).getStage(), equalTo(SegmentReplicationState.Stage.DONE));
+        assertBusy(
+            () -> assertThat(
+                response.shardSegmentReplicationStates().get(INDEX_NAME).get(0).getStage(),
+                equalTo(SegmentReplicationState.Stage.DONE)
+            )
+        );
```

Comment:
From https://github.com/opensearch-project/OpenSearch/issues/6259, the underlying cause of `testSegmentReplicationStatsWithTimeout` test failure is `org.apache.lucene.store.LockObtainFailedException` exception, on underlying store during round of segrep. I am not sure if waiting on state will resolve the error. 
The change should help fixing `testSegmentReplicationStatsResponse` though.